fastplotlib/layouts/_gridplot.py

```diff
@@ -92,12 +92,35 @@ def __init__(
                 self.shape
             )
 
+        if "names" in kwargs.keys():
+            self.names = to_array(kwargs["names"])
+            if self.names.shape != self.shape:
+                raise ValueError(f"subplot names: {self.names} must be in gridplot shape: {self.shape}")
+        else:
+            self.names = None
+
         if isinstance(controllers, str):
             if controllers == "sync":
                 controllers = np.zeros(
                     self.shape[0] * self.shape[1], dtype=int
                 ).reshape(self.shape)
 
+        # Check if 'names' and 'controllers' both have values; check if values in controller array are strings
+        if (self.names is not None) and (controllers is not None):
+            c = to_array(controllers)
+            if isinstance(c[0, 0], str):
+                idx_async_controllers = np.shape(c)[0]
```

Comment:
This is messy, use `enumerate` instead of an incrementer:

```python

positions_iter = product(range(self.shape[0]), range(self.shape[1]))
for i, position in enumerate(positions_iter):
  ...
```

Reply:
enumerate doesn't fulfill same role in this case, but since we're requiring names kwarg to have same shape now we don't need this anymore. Will rewrite